lib/yubihsm.c

```diff
@@ -118,8 +127,11 @@ static yh_rc send_authenticated_msg(yh_session *session, Msg *msg,
   memcpy(mac_buf + SCP_PRF_LEN, msg->raw, msg_len + 3 - SCP_MAC_LEN);
   msg->st.len = ntohs(msg->st.len);
   mac_buf_len = SCP_PRF_LEN + msg->st.len + 3 - SCP_MAC_LEN;
-  compute_full_mac(mac_buf, mac_buf_len, session->s.s_mac, SCP_KEY_LEN,
-                   session->s.mac_chaining_value);
+  if (compute_full_mac(mac_buf, mac_buf_len, session->s.s_mac, SCP_KEY_LEN,
+                       session->s.mac_chaining_value) != YHR_SUCCESS) {
+    DBG_ERR("compute_full_mac %s", yh_strerror(YHR_GENERIC_ERROR));
+    return YHR_GENERIC_ERROR;
```

Comment:
should we maybe propagate the rc from compute_full_mac() instead (even if it can only be YHR_SUCCESS or YHR_GENERIC_ERROR right now)?